src/AppInstallerCLICore/Workflows/ShellExecuteInstallerHandler.cpp

```diff
@@ -256,8 +256,9 @@ namespace AppInstaller::CLI::Workflow
         if (retry)
         {
             std::this_thread::sleep_for(250ms);
-            // If it fails again, let that one throw
-            std::filesystem::rename(installerPath, renamedDownloadedInstaller);
+            // If retry fails, try copying with hardlink. The file at installerPath will not be cleaned up after installation,
+            // but it is in a temp folder.
+            std::filesystem::copy(installerPath, renamedDownloadedInstaller, std::filesystem::copy_options::create_hard_links);

```

Comment:
I was thinking to make it retry every 100ms for 500ms, and only then copy.  Since the file won't get cleaned up we would really prefer the rename.

Reply:
ok.